forge-core/src/main/java/forge/card/CardFace.java

```diff
@@ -28,11 +28,12 @@ public enum FaceSelectionMethod { //
     
     private final static List<String> emptyList = Collections.unmodifiableList(new ArrayList<>());
     private final static Map<String, String> emptyMap = Collections.unmodifiableMap(new TreeMap<>());
+    private final static Set<Integer> emptySet = Collections.unmodifiableSet(new HashSet<>());
 
     private String name;
     private String altName = null;
     private CardType type = null;
-    private ManaCost manaCost = ManaCost.NO_COST;
+    private ManaCost manaCost = null;
```

Comment:
I don't know about this case, I know it wouldn't trigger `if ( manaCost == null && color == null )`
but I would prefer if we still have this "No_Cost" as Default

@tool4ever your opinion?

Reply:
The line added to `assignMissingFields` - `if ( manaCost == null ) manaCost = ManaCost.NO_COST;` - should ensure that no CardFace ever makes it out of the card reader with this null mana cost. I mainly changed it so that a variant face can know when to fall back to the cost of the parent card: `if(variant.manaCost == null) variant.manaCost = this.manaCost;`. 

I think this change would only matter if someone creates a CardFace without calling `assignMissingFields`*, which would present similar issues when the triggers and abilities lists were also all left null.

*Functional variants previously did this, but this PR cleans up that behavior too.

Reply:
my question was more if we should have kept the default "No_Cost/No_Color" thing, and we could remove the ManaCost:no_cost from the lands

@tool4ever your opinion?

Reply:
cleaning up the lands like that would make sense too

Reply:
I think we can close that issue for now (making ManaCost Required)

and then we can make a new MR to make ManaCost optional again?